### PR TITLE
Docs: document ESLint api

### PR DESCRIPTION
**Pull request by chenxsan**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

The [`ESLint`](https://github.com/eslint/eslint/blob/83cc8a6823d7729297b6814a644cca770cb99dbb/lib/api.js#L17) was exposed via this pull request https://github.com/eslint/eslint/pull/12939, but it's not documented in the latest documentation yet.

#### Is there anything you'd like reviewers to focus on?

Nothing.

### CODE REVIEW

File: docs/developer-guide/architecture.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 At a high level, there are a few key parts to ESLint:
 
 * `bin/eslint.js` - this is the file that actually gets executed with the command line utility. It's a dumb wrapper that does nothing more than bootstrap ESLint, passing the command line arguments to `cli`. This is intentionally small so as not to require heavy testing.
-* `lib/api.js` - this is the entry point of `require("eslint")`. This file exposes an object that contains public classes `Linter`, `CLIEngine`, `RuleTester`, and `SourceCode`.
+* `lib/api.js` - this is the entry point of `require("eslint")`. This file exposes an object that contains public classes `Linter`, `CLIEngine`, `ESLint`, `RuleTester`, and `SourceCode`.
```
Comment (aladdin-add) [inline docs/developer-guide/architecture.md:8]: `CLIEngine` was removed.
```suggestion
* `lib/api.js` - this is the entry point of `require("eslint")`. This file exposes an object that contains public classes `Linter`, `ESLint`, `RuleTester`, and `SourceCode`.
```

**Comment by aladdin-add** [comment]
can you rebase the master to fix the build failing, thx!

File: docs/developer-guide/architecture.md (hunk 1 above)
Comment (chenxsan) [reply docs/developer-guide/architecture.md:8]: Are you sure to remove `CLIEngine`? It's not removed until eslint v8 which is still in beta as far as I see.

Comment (aladdin-add) [reply docs/developer-guide/architecture.md:8]: yes. that PR has been merged, looks we forgot to change here. // cc @mdjermanovic

Comment (snitin315) [reply docs/developer-guide/architecture.md:8]: I think we can remove `CLIEngine`, we have already started merging v8 pull requests including documentation change. Also,`CLIEngine` has been already removed from [`nodejs-api docs`](https://github.com/eslint/eslint/blob/master/docs/developer-guide/nodejs-api.md)

Comment (nzakas) [reply docs/developer-guide/architecture.md:8]: Yes, we can remove `CLIEngine` now. Good catch!

**Review by nzakas — APPROVED** [review]
LGTM

**Review by snitin315 — APPROVED** [review]

**Pull request merged**